File: config/migrate.js
const { Client } = require('pg');
const fs = require('fs');

const client = new Client({
  host: 'db.prisma.io',
  port: 5432,
  database: 'postgres',
  user: 'f1fb86379bd7e0130f6da8596ef6d437fb4024cf24a16fd4fe72b7a7934562ce',
  password: 'sk_YtUQWogCL57ctIWQXhPid',
  ssl: { rejectUnauthorized: false },
  connectionTimeoutMillis: 10000,
  statement_timeout: 10000
});

async function runMigration() {
  try {
    console.log('Connecting to PostgreSQL at db.prisma.io:5432...');
    console.log('Timeout set to 10 seconds...');

    await client.connect();
    console.log('✓ Connected successfully!');

    const schema = fs.readFileSync(__dirname + '/schema.sql', 'utf8');
    const statements = schema
      .split(';')
      .map(s => s.trim())
      .filter(s => s.length > 0);

    console.log('Running migration...');
    for (const statement of statements) {
      console.log(`Executing: ${statement.substring(0, 50)}...`);
      await client.query(statement);
    }

    console.log('✓ Migration completed successfully!');
    process.exit(0);
  } catch (error) {
    console.error('✗ Migration failed:', error.message);
    process.exit(1);
  } finally {
    await client.end();
  }
}

runMigration();

File: config/migrate.py
#!/usr/bin/env python3
import psycopg2
import sys
import time

# Database connection parameters
conn_params = {
    'host': 'db.prisma.io',
    'port': 5432,
    'database': 'postgres',
    'user': 'f1fb86379bd7e0130f6da8596ef6d437fb4024cf24a16fd4fe72b7a7934562ce',
    'password': 'sk_YtUQWogCL57ctIWQXhPid',
    'sslmode': 'require',
    'connect_timeout': 10
}

def run_migration():
    try:
        print('Connecting to PostgreSQL at db.prisma.io:5432...')
        conn = psycopg2.connect(**conn_params)
        cursor = conn.cursor()
        print('✓ Connected successfully!')

        # Read schema file
        with open(__file__.replace('migrate.py', 'schema.sql'), 'r') as f:
            schema = f.read()

        # Split by semicolon
        statements = [s.strip() for s in schema.split(';') if s.strip()]

        print('Running migration...')
        for statement in statements:
            print(f"Executing: {statement[:50]}...")
            cursor.execute(statement)

        conn.commit()
        cursor.close()
        conn.close()
        print('✓ Migration completed successfully!')
        sys.exit(0)

    except psycopg2.OperationalError as e:
        print(f'✗ Connection failed: {e}')
        sys.exit(1)
    except Exception as e:
        print(f'✗ Migration failed: {e}')
        sys.exit(1)

if __name__ == '__main__':
    run_migration()
